### ISSUE TITLE
PointSource enforces flux as float

### ISSUE THREAD

**Issue by maxecharles** [LouisDesdoigts/dLux#264]
Related to [this `Zodiax` issue](https://github.com/LouisDesdoigts/zodiax/issues/35#issue-2007270832), where the gradients of a `float` returns `None`.

`dl.PointSource` enforces flux to be a float.
```python
"""the dLux source code"""
class PointSource(Source):
    ...
    def __init__(...):
        # Position and Flux
        self.position = np.asarray(position, dtype=float)
        self.flux = float(flux)

        if self.position.shape != (2,):
            raise ValueError("position must be a 1d array of shape (2,).")
```
This issue was fixed by setting it to a jax array in the same fashion as the position:
```python
"""my alteration of the dLux source code"""
class PointSource(Source):
    ...
    def __init__(...):
        # Position and Flux
        self.position = np.asarray(position, dtype=float)
        self.flux = np.asarray(flux, dtype=float)

        if self.position.shape != (2,):
            raise ValueError("position must be a 1d array of shape (2,).")
        if self.flux.shape != ():
            raise ValueError("flux must be a scalar value.")
```

I can push this if need be or would it be better just to fix it in Zodiax?

**Comment by LouisDesdoigts** [comment]
This would be a Zodiax side fix, probably either by pre-processing the leaves into arrays or using a different equinox filter function. Will need to check how it interacts with jax.grad calls though to make a final call

**Comment by LouisDesdoigts** [comment]
Ultimately this is a zodiax/equinox-side problem.
